src/transactions/TransactionFrame.cpp

```diff
@@ -2117,11 +2119,16 @@ TransactionFrame::processRefund(AppConnector& app, AbstractLedgerTxn& ltxOuter,
     LedgerTxn ltx(ltxOuter);
     int64_t refund = refundSorobanFee(ltx, feeSource, txResult);
 
-    // TODO: handle fee event logic here
-
     meta.pushTxChangesAfter(ltx.getChanges());
     ltx.commit();
 
+    // Emit fee event. The feeCharged has already been updated with refund in
+    // `refundSorobanFee`
+    txEventManager.newFeeEvent(feeSource, txResult.getResult().feeCharged);
```

Comment:
I don't see much point in the busy-work required to just push a single event. This should be just `pushFeeEvent`. 
Also I don't think this belongs to `processRefund`. Just emit the event in the ledger manager after `processPostApply` call - this way you ensure that we're never missing it.